packages/material-ui/src/useWidth/unstable_useMediaQuery.js

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable camelcase, import/prefer-default-export */
+import React from 'react';
+import debounce from 'debounce'; // < 1kb payload overhead when lodash/debounce is > 3kb.
+import mediaQuery from 'css-mediaquery'; // Support SSR and old browsers
+
+const browser = typeof window !== 'undefined';
+
+function matches(query, valuesInput) {
+  if (browser && window.matchMedia) {
+    return window.matchMedia(query).matches;
+  }
+
+  let values = valuesInput;
+
+  if (browser && !values) {
+    values = {
+      width: window.innerWidth,
+      height: window.innerHeight,
+    };
+  }
+
+  return mediaQuery.match(query, values);
+}
+
+export function unstable_useMediaQuery(queryInput, options = {}) {
+  const {
+    noSSR = false,
+    resizeInterval = 166, // Corresponds to 10 frames at 60 Hz.
+    values,
```

Comment:
```suggestion
    matches = window.matchMedia,
```

API change:
```diff
- useMediaQuery(query, { values });
+ const matches = query => browser ? window.matchMedia(query.matches) : { matches: mediaQuery.match(query, values) }
+ useMediaQuery(query, { matches} )
```

This removes the hard dependency on the package and makes it more explicit what is actually happening. Right now the `values` option is ignored on the server while the `matches` option is always used.

Maybe even go as far as move to DI concerning `matchMedia` and expect the return value to implement `MediaQueryList`.

Reply:
I would suggest that we go in the other direction. That we use `useTheme()` and that we use `getThemeProps()` so people can juste provide the estimated viewport of the client at one location: the theme. Now, you are right, they could provide their own matcher through the theme. Yeah, I love this solution! Just need to document it, a nice tradeoff.